network/network.go

```diff
@@ -711,6 +762,21 @@ func (n *network) Dispatch() error {
 			n.log.Debug("error during server accept: %s", err)
 			return err
 		}
+
+		// We pessimistically drop an incoming connection if the remote
+		// address is found in connectedIPs, myIPs, or peerAliasIPs.
+		// This protects our node from spending CPU cycles on TLS
+		// handshakes to upgrade connections from existing peers.
+		// Specifically, this can occur when one of our existing
+		// peers attempts to connect to one our IP aliases (that they
+		// aren't yet aware is an alias).
+		addr := conn.RemoteAddr().String()
```

Comment:
I don't think RemoteAddr().String() == ip.Desc..String()
which makes he upgradeIncoming check never work..

Reply:
You'll need to do something like thii:
localPeerIP, err := utils.ToIPDesc(addr.String())

Reply:
I was just looking around to get a clear answer but got interrupted by a meeting.

In our `peer.getVersion` IP check, we just compare IP address: https://github.com/ava-labs/avalanchego/blob/54a5d522a415e60f37ad3819b764f42821f072cf/network/peer.go#L550-L562

In `network`, we store `ip.String()` in `connectedIPs, disconnectedIPs, etc` which does include the `port`: https://github.com/ava-labs/avalanchego/blob/54a5d522a415e60f37ad3819b764f42821f072cf/utils/ip.go#L57-L59

I'm not sure if `conn.RemoteAddr().String() == utils.ToIPDesc(addr.String()).String()` but to be sure I agree we should use `utils.ToIPDesc(addr.String()).String()`. Thanks!